Makefile

```diff
@@ -0,0 +1,46 @@
+# .RECIPEPREFIX +=
+CC = gcc
+SRCS = src/ftsh.c src/ftsh_functions.c
+OBJS = $(SRCS:.c=.o)
+BINS = ftsh
+LDLIBS = -lreadline `pkg-config --libs glib-2.0`
+RM = rm -rf
+LIB = libmtrie.so
+CFLAGS = -g -O2 -Wall -std=c99 -I ./include/ -I ./src/ -fPIC `pkg-config --cflags --libs glib-2.0`
+MT_SRCS = src/mtrie.c
+MT_OBJS = src/mtrie.o
+
+MATCH_SRCS = src/match.c
+MATCH_OBJS = $(MATCH_SRCS:.c=.o)
+MATCH_BINS = src/match.o
+MATCH_LIB = libmatch.so
+
+
+all: $(LIB) ftsh $(MATCH_LIB)
```

Comment:
Let's avoid having hard-coded rules (`ftsh`). You have a `BINS` variable defined on line 5, but it's not used here. Let's change that variable name to `FTSH_BIN` and use it throughout.